### DISCUSSION TITLE
How to build complete Embedded CDT IDE from source code.

### DISCUSSION THREAD

**Discussion by Ahmed-Bilal-01** [eclipse-embed-cdt/eclipse-plugins#615] [General Discussions]
Hello,

I want to build and run Embedded CDT IDE app from its source code. But I am not getting and documentation or help. I will be very thankful to get assistance from you in this regard.

**Comment by TommyMurphyTM1234** [comment]
Eclipse Embedded CDT is not an app but a set of Eclipse plugins. 

This might help in relation to building the plugins from scratch: 

- https://github.com/eclipse-embed-cdt/eclipse-plugins/blob/master/README-MAINTAINER.md

Why do you need to build them from scratch?

If you want to also build the Eclipse IDE framework from scratch then you'll need to seek that information elsewhere - in an appropriate Eclipse specific forum, for example:

- https://github.com/eclipse-platform

**Reply by ilg-ul** [reply]
Tommy is right, this project is basically a set of Eclipse plug-ins. Building them is done by a maven command, this is relatively easy. Much more difficult is to prepare a full development environment in order to run debug sessions with the plug-ins.

Building the binaries distributed by Eclipse is a different issue, I don't know how difficult it is to do it locally, since currently everything is done centralised by the Eclipse team for all projects.

**Comment by Ahmed-Bilal-01** [comment]
@TommyMurphyTM1234 , @ilg-ul,

Thanks for your response.

Actually I am working on a solution in which it is required to provide Eclipse C/C++ Embedded IDE .exe built from source code as it is my task that's why I wanted to know about it.

So, according to you both there will be requirement to build Eclipse IDE first itself and then use Eclipse C/C++ Embedded IDE plugin in it following the plugin integration process, right?

if this is the case I am unable to get a github repo and guide/documentation regarding building Eclipse main itself from its source code.

I will be very thankful if you can assist me in this scenario.

**Reply by TommyMurphyTM1234** [reply]
> So, according to you both there will be requirement to build Eclipse IDE first itself and then use Eclipse C/C++ Embedded IDE plugin in it following the plugin integration process, right?

Yes. But, again, the Eclipse IDE is comprised of a lot more than just a single executable. Just look at what ends up in the installation directory when you install the base IDE.

> if this is the case I am unable to get a github repo and guide/documentation regarding building Eclipse main itself from its source code.

Did you look at the link that I posted above?

In any case, the question of how to build the base Eclipse IDE from sources is really outside the scope of this Embedded CDT repo.

This is old and probably out of date but might help in some way even if just to find more up to date info:

- https://stackoverflow.com/questions/37885323/building-eclipse-ide-from-scratch-how-to-specify-the-target-os
